src/webgpu/web_platform/reftests/canvas_colorspace.html.ts

```diff
@@ -0,0 +1,68 @@
+import { runRefTest } from './gpu_ref_test.js';
+
+function bgraFromRgba(rgba: Uint8Array) {
+  const bgra = rgba.slice();
+  for (let i = 0; i < bgra.length; i += 4) {
+    const t = bgra[0];
+    bgra[0] = bgra[2];
+    bgra[2] = t;
+  }
+  return bgra;
+}
+
+runRefTest(async t => {
+  const device = t.device;
+
+  // prettier-ignore
+  const kRGBAData = new Uint8Array([
+    0, 255, 0, 255,
+    117, 251, 7, 255,
+    170, 35, 209, 255,
+    80, 150, 200, 255,
+  ]);
+  const kBGRAData = bgraFromRgba(kRGBAData);
+  const width = kRGBAData.length / 4;
+
+  const testData: { [id: string]: Uint8Array } = {
+    rgba8unorm: kRGBAData,
+    bgra8unorm: kBGRAData,
+  };
+
+  function createCanvas(
+    alphaMode: GPUCanvasAlphaMode,
+    format: GPUTextureFormat,
+    colorSpace: PredefinedColorSpace
+  ) {
+    const canvas = document.createElement('canvas');
+    canvas.width = width;
+    canvas.height = 1;
+    const context = canvas.getContext('webgpu') as GPUCanvasContext;
+    context.configure({
+      device,
+      format,
+      usage: GPUTextureUsage.COPY_DST | GPUTextureUsage.RENDER_ATTACHMENT,
+      alphaMode,
+      colorSpace,
+    });
+
+    const texture = context.getCurrentTexture();
+    device.queue.writeTexture(
+      { texture },
+      testData[format],
+      {
+        bytesPerRow: kRGBAData.byteLength,
+      },
+      { width: 4, height: 1 }
+    );
+    document.body.appendChild(canvas);
+  }
+
+  createCanvas('opaque', 'rgba8unorm', 'srgb');
+  createCanvas('opaque', 'rgba8unorm', 'display-p3');
+  createCanvas('opaque', 'bgra8unorm', 'srgb');
+  createCanvas('opaque', 'bgra8unorm', 'display-p3');
+  createCanvas('premultiplied', 'rgba8unorm', 'srgb');
+  createCanvas('premultiplied', 'rgba8unorm', 'display-p3');
+  createCanvas('premultiplied', 'bgra8unorm', 'srgb');
+  createCanvas('premultiplied', 'bgra8unorm', 'display-p3');
```

Comment:
An additional thought when looking at #2176: probably good to do loops over that PR's `kCanvasColorSpaces` and a new `kCanvasAlphaModes`, to help keep the test up to date with future changes

Reply:
done

Reply:
done